src/GafferScene/Unencapsulate.cpp

```diff
@@ -51,18 +49,90 @@ using namespace GafferScene;
 namespace
 {
 
-// TODO - is there a centralized version of this?
 ScenePlug::ScenePath concatScenePath( const ScenePlug::ScenePath &a, const ScenePlug::ScenePath &b )
 {
-	ScenePlug::ScenePath result = a;
-	result.reserve( result.size() + b.size() );
-	for( const std::string & n : b )
-	{
-		result.push_back( n );
-	}
+	ScenePlug::ScenePath result;
+	result.reserve( a.size() + b.size() );
+	result.insert( result.end(), a.begin(), a.end() );
+	result.insert( result.end(), b.begin(), b.end() );
 	return result;
 }
 
+class CapsuleScope : boost::noncopyable
+{
+
+	public :
+
+		CapsuleScope(
+			const Gaffer::Context *context, const ScenePlug *inPlug,
+			const ScenePlug::ScenePath &parentPath, boost::optional<const ScenePlug::ScenePath &> branchPath, bool requireCapsule, boost::optional<const InternedString &> setName = boost::optional<const InternedString &>()
+		)
+		{
+			m_object = inPlug->object( parentPath );
+			m_capsule = IECore::runTimeCast< const Capsule >( m_object.get() );
+			if( !m_capsule )
+			{
+				if( requireCapsule )
+				{
+					// Note : John requested this exception, but I've been unable to actually devise a test
+					// that would actually trigger it, because at the point where we call CapsuleScope, we
+					// have parentPath and branchPath from parentAndBranchPaths, and parentAndBranchPaths
+					// does its own checking
+					std::string branchPathStr;
+					std::string parentPathStr;
+					ScenePlug::pathToString( *branchPath, branchPathStr );
+					ScenePlug::pathToString( parentPath, parentPathStr );
+					throw IECore::Exception( "Unencapsulating path " + branchPathStr + " within " + parentPathStr + " which is not a capsule." );
```

Comment:
Ah, good point. Let's remove then. Alternatively, we could simplify the constructor further and make this more robust by removing the `requireCapsule` argument and moving it to `scene( throwIfMissing )`. That way the "I can't handle null" argument is more closely associated with the provider of null-ness. Whatever you prefer though.

Reply:
Moved to scene()